app/models/concerns/type_of_means_assessment.rb

```diff
@@ -77,6 +78,10 @@ def no_savings?
     income.has_savings == 'no'
   end
 
+  def has_frozen_assets?
+    income.has_frozen_income_or_assets == 'yes'
+  end
+
   def no_frozen_assets?
```

Comment:
This one is then only used in the income_decision_tree. Worth modifying it so we can removed this method?

`no_frozen_assets? && !summary_only? `-> !`has_frozen_assets? && !summary_only?` or whatever cleverer way of combining them there is

Reply:
I'm not sure I understand the business logic for this switch well enough to refactor the method.